src/llvm/di.rs

```diff
@@ -348,15 +348,41 @@ unsafe fn iter_operands(v: LLVMValueRef) -> impl Iterator<Item = LLVMValueRef> {
     (0..LLVMGetNumOperands(v)).map(move |i| LLVMGetOperand(v, i as u32))
 }
 
-unsafe fn iter_metadata_copy(v: LLVMValueRef) -> impl Iterator<Item = (u32, LLVMMetadataRef)> {
-    let mut count = 0;
-    let entries = LLVMGlobalCopyAllMetadata(v, &mut count);
-    (0..count).map(move |index| {
-        (
-            LLVMValueMetadataEntriesGetKind(entries, index as u32),
-            LLVMValueMetadataEntriesGetMetadata(entries, index as u32),
-        )
-    })
+struct MetadataEntries {
+    entries: *mut LLVMValueMetadataEntry,
+    count: usize,
+}
+
+impl MetadataEntries {
+    fn new(v: LLVMValueRef) -> Option<Self> {
+        if !unsafe { can_get_all_metadata(v) } {
```

Comment:
I would just do

```suggestion
        if !unsafe { !LLVMIsAGlobalObject(v).is_null() || !LLVMIsAInstruction(v).is_null()  } {
```

here inline. Or at least move these small `is_*`/`can_*` functions to `src/llvm/types/di.rs` as well.

Reply:
ack, removing can_get_all_metadata and removing more in a followup to this